tutorials/llm/llama-3/nemo2-sft-peft/nemo2-sft.ipynb

```diff
@@ -0,0 +1,632 @@
+   "source": [
+    "def logger() -> run.Config[nl.NeMoLogger]:\n",
+    "    ckpt = run.Config(\n",
+    "        nl.ModelCheckpoint,\n",
+    "        save_last=True,\n",
+    "        every_n_train_steps=10,\n",
+    "        monitor=\"reduced_train_loss\",\n",
+    "        save_top_k=1,\n",
+    "        save_on_train_epoch_end=True,\n",
+    "        save_optim_on_train_end=True,\n",
+    "    )\n",
+    "\n",
+    "    return run.Config(\n",
+    "        nl.NeMoLogger,\n",
+    "        name=\"nemo2_sft\",\n",
+    "        log_dir=\"./results\",\n",
+    "        use_datetime_version=False,\n",
+    "        ckpt=ckpt,\n",
+    "        wandb=None\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "\n",
+    "### Optimizer\n",
+    "In the following example, we will be using distributed adam optimizer, and pass in optimizer configuration through `OptimizerConfig`: \n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def adam_with_cosine_annealing() -> run.Config[nl.OptimizerModule]:\n",
+    "    opt_cfg = run.Config(\n",
+    "        OptimizerConfig,\n",
+    "        optimizer=\"adam\",\n",
+    "        lr=5e-6,\n",
+    "        adam_beta2=0.98,\n",
+    "        use_distributed_optimizer=True,\n",
+    "        clip_grad=1.0,\n",
+    "        bf16=True,\n",
+    "    )\n",
+    "    return run.Config(\n",
+    "        nl.MegatronOptimizerModule,\n",
+    "        config=opt_cfg\n",
+    "    )\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Base Model\n",
+    "We will perform SFT on top of Llama3-8b so we create a `LlamaModel` to pass to finetune API."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def llama3_8b() -> run.Config[pl.LightningModule]:\n",
+    "    return run.Config(llm.LlamaModel, config=run.Config(llm.Llama3Config8B))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### AutoResume\n",
+    "In NeMo 2.0 we can directly pass in Llama3-8b's Hugging Face ID to start SFT without manually converting it into NeMo checkpoint format like in NeMo 1.0."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def resume() -> run.Config[nl.AutoResume]:\n",
+    "    return run.Config(\n",
+    "        nl.AutoResume,\n",
+    "        restore_config=run.Config(nl.RestoreConfig,\n",
+    "            path=\"nemo://meta-llama/Meta-Llama-3-8B\"\n",
+    "        ),\n",
+    "        resume_if_exists=True,\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "### NeMo 2.0 finetun API\n",
```

Comment:
revise heading to an imperative verb, fix capitalization, fix spelling

    "### Use the NeMo 2.0 Finetune API\n",